patches/minecraft/net/minecraft/network/datasync/EntityDataManager.java.patch

```diff
@@ -0,0 +1,21 @@
+--- ../src-base/minecraft/net/minecraft/network/datasync/EntityDataManager.java
++++ ../src-work/minecraft/net/minecraft/network/datasync/EntityDataManager.java
+@@ -38,7 +38,7 @@
+ 
+     public static <T> DataParameter<T> func_187226_a(Class <? extends Entity > p_187226_0_, DataSerializer<T> p_187226_1_)
+     {
+-        if (field_190303_a.isDebugEnabled())
++        if (true || field_190303_a.isDebugEnabled()) //Forge: This is very useful for mods that register keys on classes that are not their own
+         {
+             try
+             {
+@@ -46,7 +46,8 @@
+ 
+                 if (!oclass.equals(p_187226_0_))
+                 {
+-                    field_190303_a.debug("defineId called for: {} from {}", p_187226_0_, oclass, new RuntimeException());
++                    field_190303_a.warn("defineId called for: {} from {}", p_187226_0_, oclass); //Forge: log at warn, mods should not add to classes that they don't own
++                    if (field_190303_a.isDebugEnabled()) field_190303_a.debug("Calling stacktrace: ", new RuntimeException()); //Forge: move stacktrace to debug, not needed at warn
```

Comment:
We should use an if/else here to only log one line, so that in debug mode the log is the same as before but at warn level.